src/DotNetWorker.ApplicationInsights/FunctionsTelemetryModule.cs

```diff
@@ -20,18 +27,37 @@ public void Initialize(TelemetryConfiguration configuration)
 
             _listener = new ActivityListener
             {
-                ShouldListenTo = source => source.Name.StartsWith("Microsoft.Azure.Functions.Worker"),
+                ShouldListenTo = source => source.Name.StartsWith(TraceConstants.FunctionsActivitySource),
                 ActivityStarted = activity =>
                 {
                     var dependency = _telemetryClient.StartOperation<DependencyTelemetry>(activity);
-                    dependency.Telemetry.Type = "Azure.Functions";
-                    activity.SetCustomProperty("_depTel", dependency);
+                    dependency.Telemetry.Type = DependencyTypeInProc; // Required for proper rendering in App Insights.
+                    activity.SetCustomProperty(DependencyTelemetryKey, dependency);
                 },
                 ActivityStopped = activity =>
                 {
-                    var dependency = activity.GetCustomProperty("_depTel") as IOperationHolder<DependencyTelemetry>;
-                    _telemetryClient.StopOperation(dependency);
-                    dependency?.Dispose();
+                    // Check for Exceptions events
+                    foreach (ActivityEvent activityEvent in activity.Events)
+                    {
+                        TrackExceptionTelemetryFromActivityEvent(activityEvent, _telemetryClient);
+                    }
+
+                    if (activity.GetCustomProperty(DependencyTelemetryKey) is IOperationHolder<DependencyTelemetry> dependencyHolder)
+                    {
+                        var dependency = dependencyHolder.Telemetry;
+
+                        foreach (var item in activity.Tags)
+                        {
+                            if (!dependency.Properties.ContainsKey(item.Key))
+                            {
+                                dependency.Properties[item.Key] = item.Value;
+                            }
+                        }
+
+                        dependency.Success = activity.Status != ActivityStatusCode.Error;
+                        _telemetryClient.StopOperation(dependencyHolder);
```

Comment:
I think we will need to do some additional manual work on stop.

1. Setting of `.Success` - I don't think `StopOperation` will do that.
2. Update `Operation.Name` if the user has changed `Activity.DisplayName`.

Any others?

Reply:
`Success` is set directly above the call to `StopOperation`. Or was there another `Success` you were referring to?

DisplayName needs some work from the host to really get right, which is why I haven't touched it yet. I see it as additive in the future. I've filed https://github.com/Azure/azure-functions-host/issues/9150 to help us with this. Will try to get this in soon so we can take advantage of it.

Reply:
Ah woops, missed the `Success` setting. Thanks!